consensus/safety-rules/src/safety_rules.rs

```diff
@@ -98,6 +106,32 @@ impl<T: Payload> SafetyRules<T> {
         }
     }
 
+    /// This reconciles the key pair of a validator signer with a given validator set
+    /// during epoch changes.
+    fn reconcile_key(&mut self, epoch_state: &EpochState) -> Result<(), Error> {
+        let validator_signer = self.signer()?;
+        if let Some(expected_key) = epoch_state
+            .verifier
+            .get_public_key(&validator_signer.author())
+        {
+            let curr_key = validator_signer.public_key();
+            if curr_key != expected_key {
+                let consensus_key = self
+                    .persistent_storage
+                    .consensus_key_for_version(expected_key)
```

Comment:
probably can remove the other api that gets the consensus key without version?

Reply:
That one will be used when instantiating a new SafetRule instance, which passes only an author address and a PersistentStorage struct. At the time the verifier is still None hence cannot get the public key for the signer or get a consensus key with version.

Reply:
pointer? I don't think we should be initializing a key / signer outside of this block.

Reply:
This is fixed in latest commit